tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_connector.py

```diff
@@ -0,0 +1,232 @@
+
+        return DatastoreProfilePostgreSQL.from_dsn(
+            dsn=connection_string, profile_name="test_profile"
+        )
+
+    def _force_drop_database(self, admin_connection, database_name):
+        """Force drop a database by terminating all connections first.
+
+        Silently ignores all errors to ensure cleanup doesn't fail tests.
+        """
+        from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connection import (
+            Statement,
+        )
+
+        # Terminate all connections to the database
+        with contextlib.suppress(Exception):
+            admin_connection.run(
+                query=Statement(
+                    sql="""
+                    SELECT pg_terminate_backend(pid)
+                    FROM pg_stat_activity
+                    WHERE datname = %s AND pid <> pg_backend_pid()
+                    """,
+                    parameters=(database_name,),
+                )
+            )
+        # Now drop the database
+        with contextlib.suppress(Exception):
+            admin_connection.run(
+                statements=[f'DROP DATABASE IF EXISTS "{database_name}"']
+            )
+
+    @contextlib.contextmanager
+    def _config_context(self, system_id=None, auto_create=None):
+        """Context manager that saves and restores mlrun config.
+
+        Always restores config even if test fails.
+        """
+        import mlrun
+
+        original_system_id = mlrun.mlconf.system_id
+        original_auto_create = (
+            mlrun.mlconf.model_endpoint_monitoring.tsdb.auto_create_database
+        )
+
+        try:
+            if system_id is not None:
+                mlrun.mlconf.system_id = system_id
+            if auto_create is not None:
+                mlrun.mlconf.model_endpoint_monitoring.tsdb.auto_create_database = (
+                    auto_create
+                )
+            yield
+        finally:
+            mlrun.mlconf.system_id = original_system_id
+            mlrun.mlconf.model_endpoint_monitoring.tsdb.auto_create_database = (
+                original_auto_create
+            )
+
+    def test_auto_create_database_creates_new_database(
+        self, project_name, test_system_id, profile, admin_connection
+    ):
+        """Test that auto_create_database=True creates a new database."""
+        from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connection import (
+            Statement,
+        )
+        from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connector import (
+            TimescaleDBConnector,
+        )
+
+        expected_db_name = f"mlrun_mm_{test_system_id}"
+        connector = None
+
+        with self._config_context(system_id=test_system_id, auto_create=True):
+            try:
+                # Create connector - should create database
+                connector = TimescaleDBConnector(
+                    project=project_name,
+                    profile=profile,
+                )
```

Comment:
If you move this out of the try, you won't need `connector = None` and `if connector:`.